Outrider/1.20.0/outrider.R

```diff
@@ -0,0 +1,253 @@
+suppressPackageStartupMessages(library("tools"))
+suppressPackageStartupMessages(library("tibble"))
+suppressPackageStartupMessages(library("readr"))
+suppressPackageStartupMessages(library("dplyr"))
+suppressPackageStartupMessages(library("stringr"))
+
+# Argument parser
+parser <- ArgumentParser(description = "Process some integers")
+parser$add_argument("query", metavar = "query_input_files", nargs = "+",
+                    help = "Files or directories containing the query input files.")
+parser$add_argument("-o","--output_path", metavar = "output_path", nargs = "+",
+                    help = "Path where output of OUTRIDER will be stored.", default="./")
+parser$add_argument("-r", "--ref", metavar = "reference_input_files", nargs = "+",
+                    help = "Files or directories containing the reference input files.")
+parser$add_argument("-p", "--pref", metavar = "prefix", nargs = "+",
+                    help = "Prefix of file.", default="gene")
+args <- parser$parse_args()
+
+
+read_input_files <- function(input){
+  sampleIDs <- c()
+  count_tables <- lapply(input, function(f) {
+    input_ext <- tools::file_ext(f)
+    if(input_ext == "Rds") {
+      rse <- readRDS(f)  # RangedSummarizedExperiment
+      sampleIDs <- append(sampleIDs, colnames(rse))
+      return(as_tibble(rownames_to_column(as.data.frame(assays(rse)$counts), var="rownames")))
+    } else if (input_ext %in% c("txt", "tsv")) {
+      count_table <- read_delim(f, show_col_types=FALSE, skip=1)
+      # col 1: samples IDs, col 9: counts
+      ct <- count_table[,c(1,9)]
+      names(ct)[1] <- "rownames"
+      sampleIDs <<- append(sampleIDs, colnames(ct)[2])
+      return(ct)
+    } else {
+      stop("Input file extension is not supported.")
+    }
+  })
+  return(list("sampleIDs"=sampleIDs, "count_tables"=count_tables))
+}
+
+
+get_input <- function(input){
+  # If directories are provided
+  if(all(sapply(input, function(x) dir.exists(x)))) {
+    retrieved_files <- sapply(input, function(d){
+      list.files(path = d, pattern = "Rds|txt|tsv", full.names = TRUE, recursive = FALSE)
+    })
+  } else if(all(sapply(input, function(x) file.exists(x)))) {  # If files are provided.
+    retrieved_files <- input
+  } else {  # If both dir and files are provided, or different tyes (character, int etc)
+    stop("Input is neither dir or file.")
+  }
+  
+  return(read_input_files(retrieved_files))
+}
+
+
+merge_count_tables <- function(lst_query, lst_ref){
+  # merge count tables together.
+  lst_count_tables <- c(lst_query, lst_ref)
+  all_counts <- lst_count_tables %>%
+    Reduce(function(dtf1,dtf2) dplyr::full_join(dtf1, dtf2,by="rownames"), .)
+}
+
+
+filter_expression <- function(ods, query, prefix){
+  if(grepl("gene", prefix)){
+    ##FOR GENE LEVEL
+    ods <- filterExpression(ods, fpkmCutoff = 1, minCounts = FALSE, filterGenes = FALSE, gtfFile="/hpc/diaggen/data/databases/gencode/gencode.v44.primary_assembly.basic.annotation.gtf")
```

Comment:
It may be better to fill this path as a parameter or via a config file. The path may change and to include that change a new release of the software must be made. (or worse: this line gets overlooked at all)